File: train.js

/* F-TASK (NodeJS)

Shunday findDoublers function tuzing, unga faqat bitta string argument pass bolib, agar stringda bir hil harf qatnashgan bolsa
true, qatnashmasa false qaytarishi kerak.
MASALAN: getReverse("hello") return true return qiladi 
*/

// Masalani yechimi:

function getReverse(task) {
    const misol = task.split("");

    for (let i = 0; i < misol.length; i++) {
        if (misol.filter(harf => harf === misol[i]).length > 1) {
            return true;
        }
    }
    return false
}
const result = getReverse("hello")
console.log(result)









/*
E-TASK (NodeJS)

Shunday function tuzing, u bitta string argumentni qabul qilib osha stringni teskari qilib return qilsin.
MASALAN: getReverse("hello") return qilsin "olleh"
*/

// Masalani yechimi:
// node train.js

// function getReverse(teskari) {
//     return teskari.split("").reverse().join("");
// };
// console.log(getReverse("hello"));






/* D-TASK (NodeJS)

Shunday function tuzingki unga integerlardan iborat array pass bolsin va function bizga osha arrayning eng katta
qiymatiga tegishli birinchi indexni qaytarsin.
MASALAN: getHighestIndex([5, 21, 12, 21, 8]) return qiladi 1 sonini. */

// Masalani yechimi:


// function getHighestIndex(raqam) {
//     let katta = Math.max(...raqam);
//     const index = raqam.indexOf(katta)
//     return index
// }

// const result = getHighestIndex([5, 21, 12, 21, 8])
// console.log(result)




/*
C-TASK (NodeJS)

Shunday function tuzing, u 2ta string parametr ega bolsin, hamda agar har ikkala string
bir hil harflardan iborat bolsa true aks holda false qaytarsin
MASALAN checkContent("mitgroup", "gmtiprou") return qiladi true;
*/

// Masalani yechimi:

// function checkContent(birinchi, ikkinchi) {
//     let birinchiSorted = birinchi.split("").sort().join("");
//     let ikkinchiSorted = ikkinchi.split("").sort().join("");

//     return birinchiSorted === ikkinchiSorted;
// }

// console.log(checkContent("mitgroup", "gmtiprou"));





/*
B-TASK (NodeJS)

Shunday function tuzing, u 1ta string parametrga ega bolsin, hamda osha stringda qatnashgan raqamlarni sonini bizga return qilsin.
MASALAN countDigits("ad2a54y79wet0sfgb9") 7ni return qiladi.
*/

// Masalani yechimi:

// function countDigits(data) {
//     return data.split("")
//         .filter(filterred => filterred >= "0" && filterred <= "9")
//         .length;
// }


// console.log(countDigits("ad2a54y79wet0sfgb9"));

// ====================================================================
/* A-TASK:

Savol: 2 parametrli function tuzing, hamda birinchi parametrdagi letterni ikkinchi parametrdagi sozdan qatnashga sonini return qilishi kerak boladi.
MASALAN countLetter("e", "engineer") 3ni return qiladi.
*/


// Masalani yechimi:

// function countLetter(letter, word, callback) {
//     let count = 0;
//     if (typeof letter !== "string") callback("insert only letter", null);
//     else {
//         for (let i = 0; i < word.length; i++) {
//             if (letter === word[i]) {
//                 count++
//             }
//         }
//     }
//     callback(null, count);
// };

// countLetter("a", "Mashaqqat", (err, data) => {
//     if (err) console.log("ERROR:", err);
//     else {
//         console.log("data:", data);
//     }
// });

File: train.py

'''
O-TASK (PYTHON)

Shunday function yozing, u har xil valuelardan iborat array qabul qilsin va List ichidagi sonlar yigindisini hisoblab chiqqan javobni qaytarsin.
MASALAN: calculate_summary([10, "10", {son: 10}, true, 35]) return 45
'''


def calculate_summary(data):
    total = [x for x in data if isinstance(x, int) and not isinstance(x, bool)]
    return sum(total)


result = calculate_summary([10, "10", {"son": 10}, True, 35])
print(result)


# '''
# M-TASK (PYTHON)

# Shunday function yozing, u string qabul qilsin va string palindrom yani togri oqilganda ham, orqasidan oqilganda ham bir hil oqiladigan soz ekanligini aniqlab boolean qiymat qaytarsin.
# MASALAN: palindrom_check("dad") return True;  palindrom_check("son") return False;
# '''

# # Masalaga Yechim:


# def palindrom_check(text):
#     return text == text[::-1]


# print(palindrom_check("dad"))
# print(palindrom_check("son"))


# '''
# K-TASK (PYTHON)

# Shunday function yozing, u string qabul qilsin va string ichidagi eng uzun sozni qaytarsin.
# MASALAN: find_longest("I come from Uzbekistan") return "Uzbekistan"
# '''

# Masalaga Yechim:


# def find_longest(data):
#     list = data.split()
#     return max(list, key=lambda x: len(x))


# print(find_longest("I come from Uzbekistan"))


# I-TASK (PYTHON)

# Shunday function tuzing, unga string argument pass bolsin. Function ushbu agrumentdagi digitlarni yangi stringda return qilsin
# MASALAN: get_digits("m14i1t") return qiladi "141"

# Masalaga Yechim:


# def get_digits(data):
#     result = [x for x in data if x.isdigit()]
#     return "".join(result)


# print(get_digits("m14i1t"))


# G-TASK (PYTHON)

# Shunday function tuzingki unga integerlardan iborat array pass bolsin va function bizga osha arrayning eng katta qiymatiga tegishli birinchi indexni qaytarsin.
# MASALAN: get_highest_index([5, 21, 12, 21, 8]) return qiladi 1 sonini.

# Masalaga Yechim:

# def get_highest_index(numbers):
#     return numbers.index(max(numbers))


# print(get_highest_index([5, 21, 12, 21, 8]))
